execution/executor-benchmark/src/transaction_generator.rs

```diff
@@ -441,6 +451,124 @@ impl TransactionGenerator {
         }
     }
 
+    fn get_connected_grps_transfer_indices(
+        rng: &mut StdRng,
+        num_signer_accounts: usize,
+        block_size: usize,
+        connected_tx_grps: usize,
+    ) -> Vec<(usize, usize)> {
+        let num_accounts_per_grp = num_signer_accounts / connected_tx_grps;
+        // TODO: handle when block_size isn't divisible by connected_tx_grps; an easy
+        //       way to do this is to just generate a few more transactions in the last group
+        let num_txns_per_grp = block_size / connected_tx_grps;
+
+        if num_txns_per_grp >= num_accounts_per_grp {
+            panic!(
+                "For the desired workload we want num_accounts_per_grp ({}) > num_txns_per_grp ({})",
+                num_accounts_per_grp, num_txns_per_grp);
+        } else if connected_tx_grps > block_size {
+            panic!(
+                "connected_tx_grps ({}) > block_size ({}) cannot guarantee at least 1 txn per grp",
+                connected_tx_grps, block_size
+            );
+        }
+
+        (0..connected_tx_grps)
+            .flat_map(|grp_idx| {
+                let accounts_start_idx = grp_idx * num_accounts_per_grp;
```

Comment:
optional, but it would be even better if the account grouping is also random for every block...

Reply:
technically, I was thinking, what does random mean here ? I think even the `unused_indices.shuffle(rng);`is probably not needed.

If we are given a pool of addresses, they are random by default (since the address values don't matter for individual txns I suppose). Only after we touch an address (use it as a sender or receiver), the usage of the rest may have to be randomized.

Reply:
ideally no recognizable pattern other than the specified constraints.

I think the current implementation is almost pattern-free. One example of the pattern (under some params):
- if accounts[0] appears, it always comes very early in the block.
- if accounts[n-1] appears, it always comes very late in the block.
- a transfer of accounts[0] -> accounts[n-1] is impossible.

Reply:
But aren't accounts[0] and accounts[n-1] themselves random to begin with ?
From execution benchmark perspective, is there a difference if transfer from accounts[0] -> accounts[n-1] not possible vs transfer from accounts[0] -> accounts[1] not possible ? Values of accounts[0], accounts[1], accounts[n-1] are all random. What might matter is how there are connected wrt txns, or how many times an account sends/receives.

Reply:
Right, For implementing connected components the pattern doesn't matter. But we still want to avoid it if possible - otherwise we have the small risk of slowly overfitting our partitioner to this pattern.

Reply:
I think a way to avoid this pattern is to shuffle the account list at the beginning of each block generation. Can be implemented as generating a random permutation p=[5,2,0,1,3,4...] of all account indices, and replace accounts[I] with accounts[p[I]]

Reply:
Still, this is optional and we can do it later.

Reply:
@zjma - Actually this is a very good point and I think we need to resolve this.  So in the current implementation, let's say we have a block size of 10 and number of connected component is 5 - this pattern ensures that all conflicting txns are close to each other. For example, txn1 won't ever conflict with txn10, which will lead us to optimize partitioning for this pattern specifically. @manudhundi - To avoid this problem, how about shuffling the txn indices at the end before returning from this function?

Reply:
@sitalkedia you are saying we shuffle at the end of the block where conflicting txns are not close to each other.
@zjma you are saying we shuffle account addresses at the beginning to avoid risks of overfitting the partitioner.

Will do both